File: gatekeeper/internal/app-proxy/dto/interview_message_dto.go
package dto

type AppInterviewMessageDto struct {
	ID                     string `json:"id"`
	ContentText            string `json:"content_text"`
	ContentTranslationText string `json:"content_translation_text"`
	TipsText               string `json:"tips_text"`
	Role                   string `json:"role"`
	Type                   string `json:"type"`
	IsLastMessage          bool   `json:"is_last_message"`
	CreatedAt              string `json:"created_at"`
}

File: web-client/dto/interview-message-dto.ts
export enum InterviewMessageRole {
  User = "user",
  Interviewer = "interviewer",
  System = "system",
}

export enum InterviewMessageType {
  Default = "default",
  Error = "error",
}

export interface InterviewMessageDto {
  id: string;
  content_text: string;
  content_translation_text: string;
  tips_text: string;
  role: InterviewMessageRole;
  type: InterviewMessageType;
  is_last_message: boolean;
}
